### DISCUSSION TITLE
Boolean operations without open edges

### DISCUSSION THREAD

**Discussion by SlackerMalakor** [pyvista/pyvista#1982] [Q&A] [answered]
I have been working with PyVista a lot lately and think it is great! Although, I’ve been having a lot of problems getting boolean operations to work. A requirement for the mesh that I’m working on is that it remains manifold after the operations, but no matter what I try, open edges appear where they should not, see screenshots below.

Mesh with open edges visualized:
https://i.gyazo.com/9e94852443f44912ca091615822f6061.png

Mesh details before boolean op:
https://i.gyazo.com/08501f842a9dc00fdcca55d72e82fb6b.png

Mesh details after boolean op:
https://i.gyazo.com/e0833dfad486de1a320ef81f26d35787.png

The test code:
https://i.gyazo.com/75bfadf20739843843681b87a927c085.png

Is there any easy solution I’m overlooking here? I would appreciate any help/pointers you would be kind enough to provide, thanks in advance!

**Comment by akaszynski** [comment, marked as answer]
Can you try to repair the mesh after the boolean operation using [pymeshfix](https://github.com/pyvista/pymeshfix)?

**Reply by SlackerMalakor** [reply]
Thank you for the quick reply!

While this solved the issue with the open edges, it deformed the mesh where those edges were. Am I not using pymeshfix correctly?

Mesh after repair:
https://i.gyazo.com/e56c1fa8f363e28faf91bbad5f687a21.png

New code with pymeshfix:
https://i.gyazo.com/00cbfaabe918b1f2cfb4f89a3b628f94.png

EDIT: I set repair to verbose, here's the log if that helps:

https://i.gyazo.com/228895ef033b9362e676228346796479.png

**Reply by akaszynski** [reply]
> While this solved the issue with the open edges, it deformed the mesh where those edges were. Am I not using pymeshfix correctly?

This is a common problem with `pymeshfix` and I don't think there's a way to resolve it automatically.

In general, maintaining a manifold mesh through boolean operations is quite difficult. You might try adjusting the tolerance in the boolean operations to try to reduce (or eliminate) the hole created by the cut.
